tests/new_core/test_data_access.py

```diff
@@ -0,0 +1,399 @@
+    mock_esm_catalog.search.return_value = mock_subset
+
+    ds1 = xr.Dataset(
+        data_vars={"tasmax": (["time"], [1.0, 2.0])},
+        coords={"time": [0, 1]},
+        attrs={"source_id": "MODEL1", "experiment_id": "historical"},
+    )
+    ds2 = xr.Dataset(
+        data_vars={"tasmax": (["time"], [3.0, 4.0])},
+        coords={"time": [0, 1]},
+        attrs={"source_id": "MODEL2", "experiment_id": "historical"},
+    )
+    mock_subset.to_dataset_dict.return_value = {"dataset1": ds1, "dataset2": ds2}
+
+    # Boundary catalog
+    mock_boundary_catalog = MagicMock()
+    mock_boundaries = MagicMock()
+    boundary_dict = {
+        "none": {"entire domain": 0},
+        "states": {"NV": 28, "CA": 16},
+        "CA watersheds": {
+            "Antelope-Fremont Valleys": 11,
+            "Aliso-San Onofre": 128,
+        },
+        "lat/lon": {"coordinate selection": 0},
+        "CA counties": {"Alameda County": 54, "Alpine County": 25},
+        "CA Electric Load Serving Entities (IOU & POU)": {
+            "San Diego Gas & Electric": 3,
+            "Pacific Gas & Electric Company": 2,
+        },
+        "CA Electricity Demand Forecast Zones": {
+            "LADWP Coastal": 0,
+            "Greater Bay Area": 1,
+        },
+        "CA Electric Balancing Authority Areas": {"BANC": 0, "CALISO": 1},
+    }
+    mock_boundaries.boundary_dict.return_value = boundary_dict
+
+    # Stations dataframe
+    mock_stations_df = pd.DataFrame(
+        {
+            "LON_X": [-120.0, -121.0],
+            "LAT_Y": [35.0, 36.0],
+            "station_name": ["Station1", "Station2"],
+        }
+    )
+
+    return mock_esm_catalog, mock_boundary_catalog, mock_boundaries, mock_stations_df
+
+
+@pytest.fixture
+def mock_data_catalog_and_objs() -> Tuple:
+    """Fixture to provide a mock DataCatalog instance along with its dependencies."""
+    mock_esm_catalog, mock_boundary_catalog, mock_boundaries, mock_stations_df = (
+        make_mock_objects()
+    )
+
+    with patch(
+        "climakitae.new_core.data_access.data_access.intake.open_esm_datastore",
+        return_value=mock_esm_catalog,
+    ) as mock_open_esm, patch(
+        "climakitae.new_core.data_access.data_access.intake.open_catalog",
+        return_value=mock_boundary_catalog,
+    ) as mock_open_catalog, patch(
+        "climakitae.new_core.data_access.data_access.read_csv_file",
+        return_value=mock_stations_df,
+    ):
+        # Reset singleton for clean test state
+        DataCatalog._instance = UNSET
+        catalog_instance = DataCatalog()
+
+        with patch.object(
+            type(catalog_instance),
+            "boundaries",
+            new_callable=PropertyMock,
+            return_value=mock_boundaries,
+        ):
+            yield catalog_instance, mock_esm_catalog, mock_boundary_catalog, mock_stations_df, mock_open_esm, mock_open_catalog
+
+
+@pytest.fixture
+def mock_data_catalog(mock_data_catalog_and_objs: Tuple) -> DataCatalog:
```

Comment:
I recommend you change the name of this function, or change the name of all your `mock_data_catalog` variables in the test class below. This is called shadowing can produce unexpected behavior.